tests/test_extract.py

```diff
@@ -0,0 +1,159 @@
+    longitude = fits.Column('LONGITUDE', format='1E', unit='degree', disp='F10.6', coord_type='RA---TAN',
+                            coord_unit='angstrom', coord_ref_point=1.0, coord_ref_value=-999.0,
+                            array=np.random.rand(10, 1))
+    latitude = fits.Column('LATITUDE', format='1E', unit='degree', disp='F10.6', coord_type='ANGLE', coord_unit='deg',
+                           coord_ref_point=1.0, coord_ref_value=-999.0, array=np.random.rand(10, 1))
+    sun_alt = fits.Column('SUN_ALT', format='1E', unit='degree', disp='F6.2', array=np.random.rand(10, 1))
+    sun_zd = fits.Column('SUN_ZD', format='1E', unit='degree', disp='F6.2', array=np.random.rand(10, 1))
+    target_alt = fits.Column('TARGET_ALT', format='1E', unit='degree', disp='F6.2', array=np.random.rand(10, 1))
+    radial_vel = fits.Column('RADIAL_VEL', format='1E', unit='km /s', disp='F7.5', array=np.random.rand(10, 1))
+    shift1 = fits.Column('SHIFT1', format='1E', unit='pixel', disp='F7.3', array=np.random.rand(10, 1))
+    ly_alpha = fits.Column('LY_ALPHA', format='1E', unit='count /s', disp='G15.6', array=np.random.rand(10, 1))
+    oi_1305 = fits.Column('OI_1304', format='1E', unit='count /s', disp='G15.6', array=np.random.rand(10, 1))
+    oi_1356 = fits.Column('OI_1356', format='1E', unit='count /s', disp='G15.6', array=np.random.rand(10, 1))
+    darkrate = fits.Column('DARKRATE', format='1E', unit='count /s /pixel', disp='G15.6', array=np.random.rand(10, 1))
+    col_defs = fits.ColDefs(
+        [time, longitude, latitude, sun_alt, sun_zd, target_alt, radial_vel, shift1, ly_alpha, oi_1305, oi_1356,
+         darkrate])
+    time_hdu = fits.BinTableHDU.from_columns(col_defs)
+    time_hdu.name = "TIMELINE"
+    # create an hdu list
+    hdu_list = fits.HDUList([prim_hdu, hdu, gti_hdu, time_hdu])
+    # write the hdu to a fits file
+    if os.path.exists(file):
+        os.remove(file)
+    hdu_list.writeto(file)
+    hdu_list = fits.open(file)
+    return hdu_list
+
+
+# generate_fits_file("test.fits")
+
+def test_get_columns():
+    """
+    Test if the function is returning the right column fields
+    """
+    # Setup
+    test_data = generate_fits_file("Output/lbgu17qnq_corrtag_a.fits")
+    dt = test_data[1].data
+    detector = "FUV"
+    # detector = "NUV"
+
+    # Truth actual values
+    # Testing for FUV
+    xfull = dt.field("xfull")
+    # xfull = dt.field("xdopp") not necessary for current detector
+    if cosutil.findColumn(dt, "yfull"):
+        yfull = dt.field("yfull")
+    else:
+        yfull = dt.field("ycorr")
+    dq = dt.field("dq")
+    epsilon = dt.field("epsilon")
+
+    # Test function
+    (xf, yf, dq2, epsilon2) = extract.getColumns(test_data, detector)
+
+    # Verify
+    np.testing.assert_array_equal(xfull, xf)
+    np.testing.assert_array_equal(yfull, yf)
+    np.testing.assert_array_equal(dq, dq2)
+    np.testing.assert_array_equal(epsilon, epsilon2)
```

Comment:
This is a slightly confusing function, but I think we're missing a couple of tests here. 

For a file that has the column "xfull", xf = dt.field("xfull")
For a file that doesn't have the column "xfull", xf = dt.field("xdopp")

For a file that has the column "yfull", yf =  dt.field("yfull")
for a file that doesn't have the column "yfull", yf = dt.field("ycorr") (for FUV) and yf = dt.field("rawy") (for NUV)

For any file: 
- dq2 = dt.field("dq")
- epsilon2 = dt.field("epsilon")

So I think you should have 2 files to test for these assertions for the FUV:
1. A file that has xfull and yfull
2. A file that has xdopp and ycorr (no xfull/yfull)

And a test for the NUV, which would be the same file as 2, but passing in "NUV" instead of "FUV"